core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageAutoFeatureStep.java

```diff
@@ -345,6 +361,78 @@ public void write(String s, byte[] bytes) {
                             farray);
                 }
             }
+
+            if (entry.getValue().serialization) {
+                //any serialization requires serializationFeature to register constructor accessors
+                if (serializationSupport == null) {
+
+                    MethodCreator requiredFeatures = file.getMethodCreator("getRequiredFeatures", "java.util.List");
+                    TryBlock requiredCatch = requiredFeatures.tryBlock();
+
+                    //serialization class
+                    ResultHandle serializationFeatureClass = requiredCatch
+                            .loadClass("com.oracle.svm.reflect.serialize.hosted.SerializationFeature");
+                    ResultHandle requiredFeaturesList = requiredCatch.invokeStaticMethod(
+                            ofMethod("java.util.Collections", "singletonList", List.class, Object.class),
+                            serializationFeatureClass);
+
+                    requiredCatch.returnValue(requiredFeaturesList);
+                }
+                serializationSupport = tc.invokeStaticMethod(
```

Comment:
There seems to a lot of code here that is generated for each entry, when it should just be run once. We don't want to generate lots of code as there can potentially be lots of entries which can cause problems with the method size limitations. Can you move this code into a function that gets called for each entry rather than duplicating it each time.

Reply:
Thanks for the comment. I'll look into it and try to fix it.